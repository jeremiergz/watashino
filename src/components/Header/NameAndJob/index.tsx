import Heading from 'components/Heading';
import Image from 'components/Image';
import Link from 'components/Link';
import FlexBox from 'components/primitives/FlexBox';
import { graphql, useStaticQuery } from 'gatsby';
import React from 'react';
import { Routes } from 'utils/Routes';

const NameAndJob: React.FC = () => {
  const { dataJson, file } = useStaticQuery(graphql`
    query HeaderData {
      dataJson {
        jobTitle
      }
      file(relativePath: { eq: "profile-pic.png" }) {
        childImageSharp {
          fixed(height: 80, quality: 100, width: 80) {
            ...GatsbyImageSharpFixed_withWebp
          }
        }
      }
    }
  `);
  return (
    <Link
      alignItems="center"
      display="flex"
      flexDirection={{ _: 'column', tablet: 'row' }}
      justifyContent={{ _: 'center', tablet: 'flex-start' }}
      textAlign={{ _: 'center', tablet: 'left' }}
      to={Routes.aboutMe}
    >
      <Image fixed={file.childImageSharp.fixed} imgStyle={{ height: 80, width: 80 }} />
      <FlexBox flexDirection="column" justifyContent="center" marginLeft={{ tablet: 2 }}>
        <Heading color="primary" fontWeight="bold" letterSpacing={-1} margin={0} padding={2} variant="h3" zIndex={1}>
          Jeremie Rodriguez
        </Heading>
        <Heading
          fontWeight="semi-bold"
          margin={0}
          marginTop={-2}
          paddingBottom={2}
          paddingTop={0}
          paddingX={2}
          variant="h4"
        >
          {dataJson.jobTitle}
        </Heading>
      </FlexBox>
    </Link>
  );
};

NameAndJob.displayName = 'NameAndJob';

export default NameAndJob;
