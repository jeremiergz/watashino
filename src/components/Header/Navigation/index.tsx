import Link from 'components/Link';
import FlexBox from 'components/primitives/FlexBox';
import { useTheme } from 'components/providers/ThemeProvider';
import SVG from 'components/SVG';
import BookIcon from 'components/svgs/icons/Book';
import DescriptionIcon from 'components/svgs/icons/Description';
import HouseIcon from 'components/svgs/icons/House';
import MoodIcon from 'components/svgs/icons/Mood';
import { graphql, useStaticQuery } from 'gatsby';
import { trackCustomEvent } from 'gatsby-plugin-google-analytics';
import React from 'react';
import { Routes } from 'utils/Routes';

const iconsMapping = {
  book: BookIcon,
  home: HouseIcon,
  mood: MoodIcon,
};

const Navigation: React.FC = () => {
  const { theme } = useTheme();
  const {
    allMarkdownRemark: { totalCount },
    allNavigationJson: { nodes: links },
    dataJson: { resumeLink },
  } = useStaticQuery<GraphQL.NavigationDataQuery>(graphql`
    query NavigationData {
      allMarkdownRemark {
        totalCount
      }
      allNavigationJson {
        nodes {
          icon
          ignoreInNavigation
          keywords
          name
          navOrder
          to
        }
      }
      dataJson {
        resumeLink
      }
    }
  `);
  const handleResumeClick = () => {
    trackCustomEvent({
      action: 'click',
      category: 'Open Resume',
    });
  };
  return (
    <FlexBox as="nav" flexWrap="wrap" justifyContent="center" marginTop={{ _: 16, tablet: 0 }}>
      {links
        .filter(link => {
          if (link.to === Routes.posts && totalCount === 0) return false;
          return !link.ignoreInNavigation;
        })
        .sort((a, b) => a.navOrder - b.navOrder)
        .map(link => (
          <FlexBox alignItems="center" flexDirection="column" justifyContent="center" key={link.name} minWidth={96}>
            <Link alignItems="center" display="flex" flexDirection="column" partiallyActive to={link.to}>
              <SVG as={iconsMapping[link.icon]} fill={theme.colors.text} />
              {link.name}
            </Link>
          </FlexBox>
        ))}
      <FlexBox alignItems="center" flexDirection="column" justifyContent="center" minWidth={96}>
        <Link
          alignItems="center"
          display="flex"
          external
          flexDirection="column"
          onClick={handleResumeClick}
          to={resumeLink}
        >
          <SVG as={DescriptionIcon} fill={theme.colors.text} />
          Resume
        </Link>
      </FlexBox>
    </FlexBox>
  );
};

Navigation.displayName = 'Navigation';

export default Navigation;
