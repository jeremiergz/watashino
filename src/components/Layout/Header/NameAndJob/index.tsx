import { graphql, useStaticQuery } from 'gatsby';
import Img, { FixedObject } from 'gatsby-image';
import React from 'react';
import styled from 'styled-components';
import theme from '../../../../theme';
import Heading from '../../../widgets/Heading';
import BaseLink from '../../../widgets/Link';

const Link = styled(BaseLink)`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    text-align: center;
    ${theme.breakpoints.minWidth.tablet} {
        flex-direction: row;
        justify-content: flex-start;
        text-align: left;
    }
`;

const Description = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    color: ${theme.colors.text};
    ${theme.breakpoints.minWidth.tablet} {
        margin-left: 8px;
    }
`;

const NameAndJobComponent = () => {
    const { dataJson, file } = useStaticQuery<GraphQL.HeaderDataQueryQuery>(graphql`
        query HeaderDataQuery {
            dataJson {
                jobTitle
            }
            file(relativePath: { eq: "profile-pic.png" }) {
                childImageSharp {
                    fixed(width: 80, height: 80) {
                        ...GatsbyImageSharpFixed
                    }
                }
            }
        }
    `);
    return (
        <>
            <Link to="/">
                <Img fixed={file.childImageSharp.fixed as FixedObject} />
                <Description>
                    <Heading
                        color={theme.colors.secondary}
                        fontWeight={theme.fonts.main.weight.bold}
                        letterSpacing={-1}
                        margin={0}
                        padding={2}
                        variant="h3"
                        zIndex={1}
                    >
                        Jeremie Rodriguez
                    </Heading>
                    <Heading
                        fontWeight={theme.fonts.main.weight.semiBold}
                        margin={0}
                        marginTop={-2}
                        paddingBottom={2}
                        paddingTop={0}
                        paddingX={2}
                        variant="h4"
                    >
                        {dataJson.jobTitle}
                    </Heading>
                </Description>
            </Link>
        </>
    );
};

NameAndJobComponent.displayName = 'NameAndJob';

export default NameAndJobComponent;
