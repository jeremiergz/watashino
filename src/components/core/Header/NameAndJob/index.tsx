import { graphql, useStaticQuery } from 'gatsby';
import Img from 'gatsby-image';
import React from 'react';
import { Routes } from '../../../../utils/Routes';
import Heading from '../../../common/Heading';
import Link from '../../../common/Link';
import Flex from '../../../primitives/Flex';

const NameAndJob = () => {
  const { dataJson, file } = useStaticQuery<GraphQL.HeaderDataQuery>(graphql`
    query HeaderData {
      dataJson {
        jobTitle
      }
      file(relativePath: { eq: "profile-pic.png" }) {
        childImageSharp {
          fixed(height: 80, width: 80) {
            ...GatsbyImageSharpFixed_withWebp
          }
        }
      }
    }
  `);
  return (
    <Link
      alignItems="center"
      display="flex"
      flexDirection={{ _: 'column', tablet: 'row' }}
      justifyContent={{ _: 'center', tablet: 'flex-start' }}
      textAlign={{ _: 'center', tablet: 'left' }}
      to={Routes.aboutMe}
    >
      <Img fixed={file.childImageSharp.fixed} imgStyle={{ height: 80, width: 80 }} />
      <Flex flexDirection="column" justifyContent="center" marginLeft={{ tablet: 2 }}>
        <Heading color="primary" fontWeight="bold" letterSpacing={-1} margin={0} padding={2} variant="h3" zIndex={1}>
          Jeremie Rodriguez
        </Heading>
        <Heading
          fontWeight="semi-bold"
          margin={0}
          marginTop={-2}
          paddingBottom={2}
          paddingTop={0}
          paddingX={2}
          variant="h4"
        >
          {dataJson.jobTitle}
        </Heading>
      </Flex>
    </Link>
  );
};

NameAndJob.displayName = 'NameAndJob';

export default NameAndJob;
