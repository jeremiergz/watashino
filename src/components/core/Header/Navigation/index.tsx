import { graphql, useStaticQuery } from 'gatsby';
import React, { ReactNode } from 'react';
import Anchor from '../../../common/Anchor';
import Link from '../../../common/Link';
import Box from '../../../primitives/Box';
import Flex from '../../../primitives/Flex';
import BookIcon from '../../../svg/Book';
import DownloadIcon from '../../../svg/Download';
import HouseIcon from '../../../svg/House';
import MoodIcon from '../../../svg/Mood';
import styled from 'styled-components';

const iconsMapping = {
  book: BookIcon,
  home: HouseIcon,
  mood: MoodIcon,
};

const LinkContainer = ({ children, ...rest }: { children: ReactNode }) => (
  <Flex alignItems="center" flexDirection="column" justifyContent="center" minWidth={96} {...rest}>
    {children}
  </Flex>
);

const Svg = styled(Box)`
  fill: ${({ theme }) => theme.svg};
`;

const Navigation = () => {
  const {
    allMarkdownRemark: { totalCount },
    allNavigationJson: { nodes: links },
    dataJson: { resumeFile },
  } = useStaticQuery<GraphQL.NavigationDataQuery>(graphql`
    query NavigationData {
      allMarkdownRemark {
        totalCount
      }
      allNavigationJson {
        nodes {
          icon
          ignoreInNavigation
          keywords
          name
          navOrder
          to
        }
      }
      dataJson {
        resumeFile
      }
    }
  `);
  return (
    <Flex as="nav" marginTop={{ _: 16, tablet: 0 }}>
      {links
        .filter(link => {
          if (link.to === '/posts' && totalCount === 0) return false;
          return !link.ignoreInNavigation;
        })
        .sort((a, b) => a.navOrder - b.navOrder)
        .map(link => (
          <LinkContainer key={link.name}>
            <Link alignItems="center" display="flex" flexDirection="column" partiallyActive to={link.to}>
              <Svg as={iconsMapping[link.icon]} height={32} width={32} />
              {link.name}
            </Link>
          </LinkContainer>
        ))}
      <LinkContainer>
        <Anchor alignItems="center" display="flex" download flexDirection="column" href={resumeFile}>
          <Svg as={DownloadIcon} height={32} width={32} />
          Resume
        </Anchor>
      </LinkContainer>
    </Flex>
  );
};

Navigation.displayName = 'Navigation';

export default Navigation;
