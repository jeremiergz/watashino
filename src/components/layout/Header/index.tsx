import Conditional from '@/components/common/Conditional';
import clsx from 'clsx';
import React from 'react';

type Content = 'whereami' | 'whoami';

const navItems: { label: Content }[] = [{ label: 'whoami' }, { label: 'whereami' }];

function Header({ current, onCurrentChange }: HeaderProps): JSX.Element {
  const handleCurrentChange = (newCurrent: Content) => () => {
    if (typeof onCurrentChange === 'function' && newCurrent !== current) {
      onCurrentChange(newCurrent);
    }
  };

  return (
    <header className="fixed top-0 left-0 right-0 z-50 flex h-12 w-full max-w-5xl justify-center bg-gray-900 md:relative md:h-16 md:justify-end md:bg-transparent">
      <nav className="flex font-semibold">
        {navItems.map((item, index) => {
          const labelStart = item.label.substring(0, item.label.length - 3);
          const labelEnd = item.label.substring(item.label.length - 3, item.label.length);

          return (
            <div className={clsx('relative flex items-center', index !== 0 && 'ml-10')} key={item.label}>
              <Conditional
                condition={item.label === current}
                wrapper={children => (
                  <>
                    <span className="absolute text-sm font-bold text-cyan-600">&gt;</span>
                    {children}
                  </>
                )}
              >
                <button
                  aria-label={`Go to ${item.label}`}
                  className="cursor-pointer rounded py-1 px-4 transition hover:shadow-md hover:brightness-90"
                  onClick={handleCurrentChange(item.label)}
                >
                  <span>{labelStart}</span>
                  <span className="text-cyan-600">{labelEnd}</span>
                </button>
              </Conditional>
            </div>
          );
        })}
      </nav>
    </header>
  );
}

export type { Content };
export type HeaderProps = {
  current: Content;
  onCurrentChange?: (value: Content) => void;
};
export default Header;
