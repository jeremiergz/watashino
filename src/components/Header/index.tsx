import FlexBox from 'components/primitives/FlexBox';
import { useTheme } from 'components/providers/ThemeProvider';
import React from 'react';
import NameAndJob from './NameAndJob';
import Navigation from './Navigation';

const Header: React.FC = () => {
  const { theme } = useTheme();
  return (
    <FlexBox
      alignItems="center"
      as="header"
      flexDirection={{ _: 'column', tablet: 'row' }}
      justifyContent={{ _: 'flex-start', tablet: 'space-between' }}
      margin="auto"
      maxWidth={theme.breakpoints[5]}
      paddingBottom={32}
      paddingTop={{ _: 32, tablet: 48 }}
      paddingX={{ _: 16, tablet: 32 }}
    >
      <NameAndJob />
      <Navigation />
    </FlexBox>
  );
};

Header.displayName = 'Header';

export default Header;
