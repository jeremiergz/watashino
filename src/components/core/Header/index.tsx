import React from 'react';
import useTheme from '../../../hooks/useTheme';
import Flex from '../../primitives/Flex';
import NameAndJob from './NameAndJob';
import Navigation from './Navigation';

const Header = () => {
  const theme = useTheme();
  return (
    <Flex
      alignItems="center"
      as="header"
      flexDirection={{ _: 'column', tablet: 'row' }}
      justifyContent={{ _: 'flex-start', tablet: 'space-between' }}
      margin="auto"
      maxWidth={theme.breakpoints[5]}
      paddingX={{ _: 16, tablet: 32 }}
      paddingY={32}
    >
      <NameAndJob />
      <Navigation />
    </Flex>
  );
};

Header.displayName = 'Header';

export default Header;
