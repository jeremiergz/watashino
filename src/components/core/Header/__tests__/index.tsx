import * as Gatsby from 'gatsby';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Header from '..';
import theme from '../../../../theme';

describe('components/Layout/Header Test Suite', () => {
    beforeAll(() => {
        const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
        useStaticQuery.mockImplementation(() => ({
            allNavigationJson: {
                nodes: [
                    {
                        icon: 'icon',
                        ignoreInNavigation: false,
                        keywords: [],
                        name: 'Test',
                        navOrder: 0,
                        to: '/',
                    },
                ],
            },
            dataJson: {
                jobTitle: 'Full Stack Engineer',
            },
            file: {
                childImageSharp: {
                    fixed: {
                        base64: 'data:image/png;base64',
                        aspectRatio: 1,
                        src: '/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png',
                        srcSet:
                            '/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,\n/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,\n/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x',
                    },
                },
            },
        }));
    });

    it('matches snapshot', () => {
        const jsx = (
            <ThemeProvider theme={theme}>
                <Header />
            </ThemeProvider>
        );
        expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
    });
});
