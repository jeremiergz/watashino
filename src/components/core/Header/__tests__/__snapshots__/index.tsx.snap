// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header Test Suite matches snapshot 1`] = `
<header
  className="sc-AykKC eabPrn"
  display="flex"
>
  <a
    className="sc-AykKC sc-AykKF ivlMwz"
    href="/about-me"
    target="_blank"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "display": "inline-block",
          "height": undefined,
          "overflow": "hidden",
          "position": "relative",
          "width": undefined,
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/png;base64"
        style={
          Object {
            "height": "80px",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transitionDelay": "500ms",
            "width": "80px",
          }
        }
      />
      <picture>
        <source
          srcSet="/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x"
        />
        <img
          alt=""
          loading="lazy"
          onLoad={[Function]}
          src="/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png"
          srcSet="/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x"
          style={
            Object {
              "height": "80px",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "80px",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x\\" /><img loading=\\"lazy\\" srcset=\\"/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png 1x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/1e576/profile-pic.png 1.5x,
/static/89ee547c5db6cbf9b37a682e8b3f3f57/6cb10/profile-pic.png 2x\\" src=\\"/static/89ee547c5db6cbf9b37a682e8b3f3f57/8b091/profile-pic.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
    <div
      className="sc-AykKC hQAXLw"
      display="flex"
    >
      <h3
        className="sc-AykKD sc-AykKE ejUWqZ"
        color="primary"
        fontWeight="bold"
        letterSpacing={-1}
      >
        Jeremie Rodriguez
      </h3>
      <h4
        className="sc-AykKD sc-AykKE gAWIEk"
        fontWeight="semi-bold"
      >
        Full Stack Engineer
      </h4>
    </div>
  </a>
  <nav
    className="sc-AykKC fWKbuP"
    display="flex"
  >
    <div
      className="sc-AykKC dUqoui"
      display="flex"
    >
      <a
        className="sc-AykKC sc-AykKF cDDyWc"
        href="/"
        target="_blank"
      >
        <svg
          className="sc-AykKG falauK"
          height={32}
          viewBox="0 0 24 24"
          width={32}
          xmlns="http://www.w3.org/2000/svg"
        />
        Test
      </a>
    </div>
    <div
      className="sc-AykKC dUqoui"
      display="flex"
    >
      <a
        className="sc-AykKC sc-AykKF cDDyWc"
        display="flex"
        download={true}
        fontWeight="semi-bold"
        target="_blank"
      >
        <svg
          className="sc-AykKG falauK sc-AykKG falauK"
          height={32}
          viewBox="0 0 24 24"
          width={32}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
        Resume
      </a>
    </div>
  </nav>
</header>
`;
